src/_pytest/nodes.py

```diff
@@ -447,6 +447,8 @@ def _prunetraceback(self, excinfo):
 def _check_initialpaths_for_relpath(session, fspath):
     for initial_path in session._initialpaths:
         if fspath.common(initial_path) == initial_path:
+            if initial_path.isdir():
+                return fspath.relto(initial_path)
             return fspath.relto(initial_path.dirname)
```

Comment:
According to codecov only the `isdir` (new) path is covered: https://codecov.io/gh/pytest-dev/pytest/commit/8532756fe549edb6ecb01733f33f714cebdc35b2#D1-450

Reply:
And looking closer it shows that it can only really be a dir here, if it shares a / is the common path with `fspath`, no?

Reply:
Amended.